src/main/java/ua/lviv/iot/parkingServer/kafka/KafkaConsumer.kt

```diff
@@ -0,0 +1,49 @@
+package ua.lviv.iot.parkingServer.kafka
+
+import java.util.*
+import org.apache.kafka.clients.consumer.ConsumerConfig
+import org.apache.kafka.common.serialization.Deserializer
+import org.apache.kafka.common.serialization.StringDeserializer
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.kafka.annotation.EnableKafka
+import org.springframework.kafka.core.reactive.ReactiveKafkaConsumerTemplate
+import reactor.kafka.receiver.ReceiverOptions
+
+@EnableKafka
+@Configuration
+class KafkaConsumer {
+
+    @Value("\${spring.kafka.bootstrap-servers}")
+    private lateinit var kafkaAddress: String
+
+    @Bean
+    fun kafkaConsumerOptions(): ReceiverOptions<String, ByteArray> {
+        val basicReceiverOptions: ReceiverOptions<String, ByteArray> =
+            ReceiverOptions.create(
+                mapOf(
+                    Pair(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaAddress),
+                    Pair(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer::class.java),
+                    Pair(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ProtobufDeserializer::class.java),
+                    Pair(ConsumerConfig.GROUP_ID_CONFIG, "parking")
+                )
+            )
+        return basicReceiverOptions.subscription(Collections.singletonList("ADDED_AVAILABLE_PARKING_SPOT"))
+    }
+
+    @Bean
+    fun kafkaConsumerTemplate(
+        kafkaReceiverOptions: ReceiverOptions<String, ByteArray>
+    ): ReactiveKafkaConsumerTemplate<String, ByteArray> {
+        return ReactiveKafkaConsumerTemplate(kafkaConsumerOptions())
+    }
+}
+
+class ProtobufDeserializer() : Deserializer<ByteArray> {
+
+    override fun deserialize(topic: String, data: ByteArray): ByteArray {
+        return data
+    }
```

Comment:
usually in one topic should be published only one event, so you can create map which will match topic and proto parser. And your deserializer will return GeneratedMessage instead of byte array